### PR TITLE
Create ldap_integration.adoc

### PR DESCRIPTION
**Pull request by zer0glitch**
This is the basis for the LDAP integration.  Group Syncing and Active Directory integration are still missing.

### CODE REVIEW

**Comment by etsauer** [comment]
@zer0glitch like what i'm seeing so far. I'm still reviewing, but one thing I would like to see changed is to move this into the Smart Start playbook rather than operationalizing.

**Comment by etsauer** [comment]
@zer0glitch ok, some comments on your doc:

- Check out our style guide. specifically, the section on beginning a new document and the heading and metadata we want to see so that it can be indexed by the playbooks website build process.
- The headers should be nested sequentially.. meaning the title of the doc should be a top level header, (i.e. `= LDAP Integration`) and then step down sequentially from there.
- I would move the 'References' section to the end of the doc
- you've got a few syntax issues
 -- Numbered lists should be formatted with `1. [content...]`
 -- Code blocks should be wrapped in '----' (4 hyphens), i.e.
```
[source,bash]
----
... code ...
----
```

I have a few more comments I'll add in line.

File: playbooks/operationalizing/ldap_integration.adoc [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        - dn # the unique id of the user, this is being mapped to the dn field in ldap
+        email: 
+        - mail # users email that is being mapped
+        name: 
+        - cn  # This is the display name it is mapped to the cn(common name field in LDAP)
+        preferredUsername: 
+        - uid # this is the field that is the username when logging into open shift
+      bindDN: "" # If the ldap has anonymous bind turned off, you may need to add a username and password to search for users
+      bindPassword: "" 
+      ca: my-ldap-ca-bundle.crt # This should be provided to you
```
Comment (etsauer) [inline playbooks/operationalizing/ldap_integration.adoc]: I would clarify here that "This should be provided by your LDAP administrator"

File: playbooks/operationalizing/ldap_integration.adoc [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+We will assume we are connection to a standard ldap.  The first step is to create a query to limit the user to a group, so from above we have:
+
+[source,conf]
+(&(objectClass=posixAccount)(memberof=CN=openshift,OU=groups,DC=example,DC=com))
+
+so in english this looks like:
+
+Where object class equals posixAccount and the user is a member of the openshift group.
+
+==== Testing your LDAP queries
```
Comment (etsauer) [inline playbooks/operationalizing/ldap_integration.adoc]: This section is awesome! We've needed this piece for a long time.

**Comment by shah-zobair** [comment]
Is there any way to use encrypted password for the "bindPassword" field?

**Comment by zer0glitch** [comment]
@shah-zobair I have not found a way to do encrypted passwords yet.  Generally the bind user does not have much ability other than to authenticate against the LDAP.  This is often done with no authentication as anonymouse.

**Comment by zer0glitch** [comment]
@etsauer I have completed the recommended change I believe.  I wasn't sure what the Smart Start playbook was, so I created a directory called smart_start.

**Comment by etsauer** [comment]
@zer0glitch sorry... the smart start playbook is just the 'installation' directory. please place it there.

**Comment by etsauer** [comment]
@zer0glitch doc looks good. Only thing left is:

- Move into /playbooks/installation/
- Add a link to this guide in /playbooks/installation/index.adoc, i.e.
`link:./ldap_integration{outfilesuffix}[Integrating with LDAP & Active Directory]`

**Comment by zer0glitch** [comment]
@etsauer 
I have updated with the last two things.

**Comment by etsauer** [comment]
Looks good to me.

@trevorquinn @oybed @sabre1041 can I get a second?

**Comment by sabre1041** [comment]
@etsauer looks good to me :+1:

**Comment by oybed** [comment]
May want to mention that this is for OSEv3.1 and later (due to various file paths). Maybe we can try to standardize a way to display which versions of OSE a playbook is applicable to. If so, maybe this is a "later" task. 

Other than that, looks good to me.

**Comment by sabre1041** [comment]
@oybed maybe we could put a section at the top of the docs to denote which OSE version it is applicable for. However, I can see this become burdensome as our playbooks continue to expand. 

There has to be some happy medium

**Comment by etsauer** [comment]
@oybed @sabre1041 let's take this offline, and get whatever we decide added to the style guide. I'm going to merge

**Comment by etsauer** [comment]
@zer0glitch thanks a bunch for this!

**Pull request merged**